ir/src/main/scala/org/scalajs/core/ir/ClassKind.scala

```diff
@@ -26,6 +26,7 @@ object ClassKind {
   case object Interface extends ClassKind
   case object RawJSType extends ClassKind
   case object HijackedClass extends ClassKind
+  case object JSClass extends ClassKind
```

Comment:
I think adding `isAnyClass` to `ClassKind` might be a good idea, given how many times `x.isClass || x == ClassKind.JSClass` occurs in the codebase.